packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js

```diff
@@ -17,24 +21,23 @@ const BottomSheetSubSheet = ( {
 	showSheet,
 	isFullScreen,
 } ) => {
-	const BottomSheetScreen = ( { onMount } ) => {
-		useEffect( onMount, [] );
-		return children ?? null;
-	};
```

Comment:
My understanding from the PR description and testing is that the creation of this component _within_ the render of its parent `BottomSheetSubSheet` is the root cause of the issue, correct? 

I am not suggesting we change this PR, but I wanted to share the below alternative that is closer to the original implementation, purely for discussion's sake. It would appear merely hoisting `BottomSheetScreen` out of `BottomSheetSubSheet` does fix the original keyboard/focus issue. 

<details><summary>Alternative Approach</summary>

```diff
diff --git a/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js b/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js
index 8a4701c693..2af2c0f5da 100644
--- a/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js
+++ b/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js
@@ -6,38 +6,37 @@ import { SafeAreaView } from 'react-native';
 /**
  * WordPress dependencies
  */
-import { Children, useEffect, useContext } from '@wordpress/element';
-import {
-	createSlotFill,
-	BottomSheetConsumer,
-	BottomSheetContext,
-} from '@wordpress/components';
+import { Children, useEffect } from '@wordpress/element';
+import { createSlotFill, BottomSheetConsumer } from '@wordpress/components';
 
 const { Fill, Slot } = createSlotFill( 'BottomSheetSubSheet' );
 
+const BottomSheetScreen = ( { children, onMount } ) => {
+	useEffect( onMount, [] );
+	return children;
+};
+
 const BottomSheetSubSheet = ( {
 	children,
 	navigationButton,
 	showSheet,
 	isFullScreen,
 } ) => {
-	const { setIsFullScreen } = useContext( BottomSheetContext );
-
-	useEffect( () => {
-		if ( showSheet ) {
-			setIsFullScreen( isFullScreen );
-		}
-	}, [ showSheet, isFullScreen ] );
-
 	return (
 		<>
 			{ showSheet && (
 				<Fill>
 					<SafeAreaView>
 						<BottomSheetConsumer>
-							{ () => {
-								return children;
-							} }
+							{ ( { setIsFullScreen } ) => (
+								<BottomSheetScreen
+									onMount={ () =>
+										setIsFullScreen( isFullScreen )
+									}
+								>
+									{ children }
+								</BottomSheetScreen>
+							) }
 						</BottomSheetConsumer>
 					</SafeAreaView>
 				</Fill>
```

</details>

Reply:
Thanks for this, David! 🙇‍♀️ You're right on what I believe to be the root cause of this issue (i.e. the creation of the child component within the parent). I appreciate you sharing that alternative approach, as hoisting the component isn't something I'd have come up with by myself, so will add it to my knowledge bank going forward. 

Are there any positives or negatives to either approach? I'm definitely open to learning and updating the PR if there is anything steering you towards or away from either approach. If it's the case that they're fairly similar, with no major advantages/disadvantages to one another, I'll go ahead to merge the current PR.

Reply:
> Are there any positives or negatives to either approach? I'm definitely open to learning and updating the PR if there is anything steering you towards or away from either approach. If it's the case that they're fairly similar, with no major advantages/disadvantages to one another, I'll go ahead to merge the current PR.

No, I wouldn't say there is a distinct positive or negative with either approach. It is merely differently structured code from my perspective. 

That said, after reviewing this again, I believe [this wrapping `BottomSheetConsumer`](https://github.com/WordPress/gutenberg/blob/e53d768735bc3848f2d27c535394f904654c693e/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js#L37) may be duplicative of [the new `useContext` usage](https://github.com/WordPress/gutenberg/blob/e53d768735bc3848f2d27c535394f904654c693e/packages/components/src/mobile/bottom-sheet/sub-sheet/index.native.js#L24) and unnecessary. I.e. we could likely remove `BottomSheetConsumer` altogether. 

```jsx
// React Context accessed via Consumer with function as a child 
<BottomSheetConsumer>
	{({ setIsFullScreen }) => {
		// ...
	}}
</BottomSheetConsumer>

// React Context accessed via useContext Hook
const { setIsFullScreen } = useContext(BottomSheetContext);
```

For an explanation as to why, both of the above are accessing the React Context object. The former is using [the `Context.Consumer` approach](https://reactjs.org/docs/context.html#contextconsumer). The latter is using [the React Hook approach](https://reactjs.org/docs/hooks-reference.html#usecontext). Either is fine. I believe this project generally attempts to rely upon React Hooks now-a-days.

Currently, this PR has a lingering `BottomSheetConsumer` that is largely going unused.

Reply:
Ah, excellent, that's clear, thanks for the explanation! I've gone ahead to remove code surrounding that unnecessary `BottomSheetConsumer` in https://github.com/WordPress/gutenberg/pull/33845/commits/523239866d5b502672695af6daa30453b57d26ac and https://github.com/WordPress/gutenberg/pull/33845/commits/b75e766a615f38091b8a8dca55a792ab007f8562. I also tested and can confirm the alt text settings still work with the update.